### ISSUE TITLE
Keep manual changes to the PR body upon update (GitHub/GitLab)

### ISSUE THREAD

**Issue by realloser** [renovatebot/renovate#18915]
### What would you like Renovate to be able to do?

Manual changes to the pull request body should be retained. This works only for platforms, where hidden comments are supported and only for changes outside of the renovate specific body.

This is a followup issue created out of this [discussion: How to modify pull request body without clearing it on update](https://github.com/renovatebot/renovate/discussions/18914)

### If you have any ideas on how this should be implemented, please tell us here.

In order to retain manual changes outside of the renovate body we can use hidden tags analog to `<!--renovate:debug....-->` as placeholders.

1. In general renovate body will **always** be wrapped between placeholder for supporting platforms, which simplifies adding changes.

2. As long the hidden placeholders exists and are valid, only the content and placeholder are replaced (with 1.). If the hidden placeholders are not valid (missing parts), the current implementation takes place (replacing the full body with 1.).

Proposal for new default renovate body:
```html
<!--renovate:start-->
Renovate specific body
<!--renovate:debug....-->
<!--renovate:otherTags....-->
<!--renovate:end-->
```

Upon update only this will be replaced.

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by rarkins** [comment]
So the general idea is that this will allow users or bots to make changes to the PR content before or after Renovate's content without it getting replaced.

Ideas:
- These lines should be removed/absent for platforms which don't support hidden html comments
- We should avoid updating *every single existing PR* once this feature lands (i.e. to add these wrapper comments). Instead, it should be added over time either (a) whenever the PR content needs updating for some other reason, or (b) for all new PRs

Here's the PR struct today: https://github.com/renovatebot/renovate/blob/92eb2200fd2209ac082f52e5a0648f5f71411ece/lib/modules/platform/types.ts#L52-L57

I think this should be extended to include `preContent` and `postContent` (raw/complete). If they are present in a PR then they should be added back to the PR whenever the body needs updating.
